File: code-smells-project/middlewares/error_handler.py
import logging
from flask import jsonify

logger = logging.getLogger(__name__)


class AppError(Exception):
    def __init__(self, message, status_code=400):
        self.message = message
        self.status_code = status_code
        super().__init__(message)


def register_error_handlers(app):
    @app.errorhandler(AppError)
    def handle_app_error(error):
        return jsonify({"error": error.message}), error.status_code

    @app.errorhandler(404)
    def handle_not_found(error):
        return jsonify({"error": "Recurso não encontrado"}), 404

    @app.errorhandler(405)
    def handle_method_not_allowed(error):
        return jsonify({"error": "Método não permitido"}), 405

    @app.errorhandler(Exception)
    def handle_generic(error):
        logger.exception("Unhandled exception")
        return jsonify({"error": "Erro interno do servidor"}), 500

File: ecommerce-api-legacy/src/middlewares/error_handler.js
function errorHandler(err, req, res, next) {
    if (err.statusCode) {
        return res.status(err.statusCode).json({ error: err.message });
    }
    console.error("Unhandled error:", err);
    res.status(500).json({ error: "Erro interno do servidor" });
}

module.exports = { errorHandler };
